herd/AArch64Sem.ml

```diff
@@ -855,7 +855,7 @@ module Make
         let get_k = match kr with
         | K k -> M.unitT (V.intToV k)
         | RV(_,rO) -> read_reg_ord rO ii in
-        get_k >>= fun k -> if V.is_zero k then M.unitT() else
+        get_k >>= fun k -> if k=V.zero then M.unitT() else
```

Comment:
Using polymorphic equality (=) is not advisable in general. The test `k=V.zero` can be replaced by `V.is_var_determined k && V.is_zero k`.

Reply:
Thank you, I've changed this to your suggestion.